src/main/java/backend/FileDecoder.java

```diff
@@ -0,0 +1,65 @@
+package backend;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+public class FileDecoder {
+
+    public FileDecoder() {}
+
+    /**
+     * Returns the task equivalent from a task generating string input in pre-saved file
+     *
+     * @param tasks A task generating string that follows standard format
+     * @return Task object containing information from input string
+     */
+    public Task decode(String tasks) {
+
+        String[] args = tasks.split(" \\| ");
+        String taskType = args[0];
+        Boolean isDone = args[1].contains("X");
+        String taskDescription = args[2];
+        String taskDate = null;
+
+        // if String supplied is one with a date
+        if (args.length == 4) {
+            try {
+                SimpleDateFormat dateFormatter = new SimpleDateFormat("MMM dd yyyy");
```

Comment:
I suggest you bring 
```
                SimpleDateFormat dateFormatter = new SimpleDateFormat("MMM dd yyyy");
```
out as a global constant for Task class so you don't have to create a new SimpleDateFormat everytime decode(String tasks) is called.
This is just for efficiency though.